tests/test_browser.py

```diff
@@ -294,5 +294,13 @@ def test_post(httpbin):
     assert resp.status_code == 200 and resp.json()['form'] == data
 
 
+def test_encoding(httpbin):
+    browser = mechanicalsoup.Browser()
+    # Get HTML encoded in UTF-8 from httpbin
+    # (UTF-8 specified in HTTP headers, no meta tags)
+    page = browser.get(httpbin + "/html")
+    assert page.soup.original_encoding == 'utf-8'
+
+
```

Comment:
It's good to have a test, but that would be even better to play with `requests_mock` (see `setup_mock_browser` in `tests/utils.py`) to actually cover all cases in the `html_encoding or http_encoding` expression (both set, both unset, only one set).

Reply:
Thanks for the review. This should be fixed!